### PR TITLE
(feat) Add TUS support

### PR DESCRIPTION
**Pull request by nitriques**
This PR implement TUS protocol to be able to upload super large files to CF Stream.
It's kind of chuck-upload, but is requires the "creation" of the file before bytes can be sent.

The code should be straight forward enough. I am open to comments.

This feature has been backported to v1/craft 4 and released as `1.6.0`

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -85,14 +85,23 @@ query MyQuery {
 }
 ```
 
-8) You can also mass re-upload everything via Craft's cli
+8) You can also mass re-upload everything via Craft's cli. Please note that this creates _news_ streams.
```
Comment (f-elix) [inline README.md]:
```suggestion
8) You can also mass re-upload everything via Craft's cli. Please note that this creates _new_ streams.
```

File: src/client/CloudflareVideoStreamClient.php [hunk 2]
```diff
@@ -63,8 +63,9 @@ public function uploadVideoByUrl(string $videoUrl, string $videoName, ?string $v
     public function uploadVideoByPath(string $videoPath, string $videoFilename)
     {
         $client = new GuzzleHttp\Client();
-        $fullPath = rtrim($videoPath, '/') . '/' . $videoFilename;
-        // Guzzle might close the file for us...
+        $fullPath = $this->createSafeFullPath($videoPath, $videoFilename);
+
+        // Guzzle might? close the file for us...
```
Comment (f-elix) [inline src/client/CloudflareVideoStreamClient.php:68]:
```suggestion
        // Guzzle might close the file for us...
```

File: src/jobs/PollVideoJob.php [hunk 3]
```diff
@@ -104,9 +104,9 @@ public function execute($queue): void
         // We need to if the process is not completed or if we don't still have a mp4 url
         if (!$this->completed || !$hasMp4Url) {
             $this->setProgress($queue, 0, 'Delayed retry');
-            // Retry the job after x * 2 seconds
+            // Retry the job after x * 1.2 seconds
```
Comment (f-elix) [inline src/jobs/PollVideoJob.php]: 1.5 seconds?

**Review by f-elix — CHANGES_REQUESTED** [review]

File: src/client/CloudflareVideoStreamClient.php (hunk 2 above)
Comment (nitriques) [reply src/client/CloudflareVideoStreamClient.php:68]: That's cuz I am not sure about it.....

File: src/jobs/PollVideoJob.php (hunk 3 above)
Comment (nitriques) [reply src/jobs/PollVideoJob.php]: @f-elix See [303a7d9](https://github.com/DeuxHuitHuit/craft-cloudflare-stream/pull/14/commits/303a7d92b9cfe2191ec23de7fea725b5662d1ca8)

**Review by f-elix — APPROVED** [review]

**Pull request merged**